doc/md/versioned-migrations.md

```diff
@@ -241,3 +241,115 @@ func main() {
    }
 }
 ```
+
+## Atlas migration directory integrity file
+
+Suppose you have multiple teams develop a feature in parallel and both of them need a migration. If Team A and Team B do
+not check in with each other, they might end up with a broken migration directory since new files do not raise a merge
+conflict in a version control system like git. To prevent concurrent creation of new migration files and guard against
```

Comment:
Maybe give a scenario example? e.g. Both teams tried to add the same column => will cause a migration failure.